docs/proposal/kmesh-reliability.md

```diff
@@ -0,0 +1,164 @@
+---
+title: Kmesh reliability
+authors:
+- "@kwb0523" # Authors' GitHub accounts here.
+- "@nlgwcy"
+reviewers:
+- "@bitcoffeeiux"
+- "@hzxuzhonghu"
+- "@supercharge-xsy"
+approvers:
+- "@robot"
+
+creation-date: 2024-06-04
+
+---
+
+## Kmesh reliability
+
+### Summary
+
+This proposal describes how to ensure that the governance capability is not affected when the Kmesh is restarted.
+
+### Motivation
+
+The problem is from [issue_322](https://github.com/kmesh-net/kmesh/issues/322). After the Kmesh is restarted, the pods managed by the Kmesh are no longer taken over by the Kmesh. As a result, the governance function is affected.
+
+#### Goals
+
+In normal and abnormal restart scenarios, the data plane governance function is not affected, including:
+
+1. After the pod is restarted, it is automatically restored to the status before the restart.
+2. During the Kmesh restart, newly established links have the governance capability.
+3. After the Kmesh is restarted, the newly created link has the governance capability.
+4. The governance capability of old links is not affected by restart.
+5. When Kmesh is deleted, resources can be completely cleared to avoid residual resources.
+
+#### Non-Goals
+
+Currently, this document focuses on the Kmesh daemon restart scenario. The following scenarios are not included in the design scope:
+
+1. Upgrade the Kmesh daemon.
+
+### Proposal
+
+The persistence capability of the bpf map and the configuration restoration capability after the Kmesh restart need to be supported, including:
+
+1. When the Kmesh is restarted, the bpf map and bpf prog need to be maintained without being deleted, which is the basis for maintaining the data plane governance capability.
+2. After the Kmesh is restarted, the configuration restoration module is added to restore the governance information of the pods managed by the Kmesh on the node.
```

Comment:
Assume this is during bootstrap, kmesh populate the bpf map again with all the workload/service info